src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -22,7 +23,7 @@ public UserController(UserService userService,
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto findByEmail(@RequestParam String email) {
+    public UserResponseDto findByEmail(@RequestParam @Email String email) {
         User user = userService.findByEmail(email).orElseThrow(
                 () -> new RuntimeException("Invalid email"));
```

Comment:
You put @Email annotation and have the exception handler.
If email is not correct, you will get an appropriate exception.
So, exception message you are throwing not quite correct. But it's minor.